### ISSUE TITLE
[a11y]: NotificationPanel: Ensure interactive controls are not nested

### ISSUE THREAD

**Issue by makafsal** [carbon-design-system/ibm-products#7612]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.67.0

### React version

v17.0.1

### Automated testing tool and ruleset

Storybook built-in accessibility tab

### Description

<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/bcab8f27-a80d-4d7e-8740-a31000329319" />

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-notifications-panel-notificationspanel--default&globals=viewport:basic

### Steps to reproduce

1. Open the default storybook
2. Check the accessibility tab

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
We can probably move the close button outside `.c4p--notifications-panel__notification` element which already has a role button. We may need to have a relative wrapper and absolute position the close button as per design.

Address this issue in web-component version as well

**Comment by elycheea** [comment]
This is the approach we recommended on the WC implementation.

> Matt and I tried to quickly mock up the new behavior so that we no longer run into the current a11y issues with current nested interactions (buttons in buttons).
> 
> ![Image](https://github.com/user-attachments/assets/591c5b9c-556e-4b5b-834c-e91772291da9)
> 
> We believe that adding the wrapper will limit CSS changes to the existing classes/elements to minimize impact to the shared styles used by React. For the icon button, we can likely use parent (`>`) child selector versus sibling (`+`) selector 

 _Originally posted by @elycheea in [#7373](https://github.com/carbon-design-system/ibm-products/issues/7373#issuecomment-2855525566)_

**Comment by szinta** [comment]
> This is the approach we recommended on the WC implementation.
> 
> > Matt and I tried to quickly mock up the new behavior so that we no longer run into the current a11y issues with current nested interactions (buttons in buttons).
> > ![Image](https://github.com/user-attachments/assets/591c5b9c-556e-4b5b-834c-e91772291da9)
> > We believe that adding the wrapper will limit CSS changes to the existing classes/elements to minimize impact to the shared styles used by React. For the icon button, we can likely use parent (`>`) child selector versus sibling (`+`) selector
> 
> _Originally posted by [@elycheea](https://github.com/elycheea) in [#7373](https://github.com/carbon-design-system/ibm-products/issues/7373#issuecomment-2855525566)_

This approach works for the Web Component version, but only partially addresses accessibility issues in the React version. In React, the notification component also includes a link and a "Read more" button, which can still lead to `Nested interactive elements` violations.

As discussed, we've decided to create a [follow-up issue ](https://github.com/carbon-design-system/ibm-products/issues/8057)to refine the NotificationsPanel react component. The refinement should consider the following points:

- Create a standalone subcomponent for individual notification items
- Remove "Read more" button functionality and links from the core component
- Make the NotificationsPanel more composable
- Use the TruncatedText utility to replace the "Read more" button

**Comment by szinta** [comment]
Closing this issue as these violations are not reproducible in the Accessibility tab of React and WC storybooks and using Equal Access Checker.

**Comment by tombrunet** [comment]
This is a violation, and is listed in the "Inconclusive" section of the storybook checker and "Needs Review" in the browser extension

<img width="707" height="813" alt="Image" src="https://github.com/user-attachments/assets/79993a73-27d3-42e0-908e-e41f69f2deef" />

A screen reader user will hear:
> ["4 months ago Cluster unreachable Not able to establish connection with provided cluster. Please check your logs and memory allocation to resolve this...", button]

But will not hear about any of the semantics within that button, nor will they be able to activate the button in a button.

Note that this issue is 'needs review' because there are scenarios where roles within presentational roles don't result in actual accessibility issues. In this case, this is a violation.
